src/test/test_dynamoTable.py

```diff
@@ -0,0 +1,167 @@
+import pytest
+import boto3
+from unittest.mock import patch, Mock
+from ..main.dynamoTable import *
+from ..main.building import *
+from ..main.machine import *
+from ..main.user import *
+
+class TestDynamoTable:
+    # User Table tests ============================================================================
+    @patch('boto3.resource')
+    def test_UserPut(self, mock_boto3):
+        userTable = DynamoTable('Users')
+        userTable.table.put_item = Mock(name='put_item')
+        
+        newUser = User('userId1', 'userBuilding', 'userBuildingID1')
+        userTable.put(newUser)
+        userTable.table.put_item.assert_called_once_with(Item=(newUser))

```

Comment:
Since you've switched from using a dictionary to using an object, you're going to need to call `vars` on newUser, newMachine, etc.